rmc/utils/constraint.py

```diff
@@ -1310,45 +1318,350 @@ def create_context_with_oe(
+        | (hl.is_missing(ht.stop_aa) & ht.transcript_stop)
+    )
+    miss_start_stop_ht = miss_start_stop_ht.checkpoint(
+        f"{TEMP_PATH_WITH_FAST_DEL}/rmc/transcript_start_stop_missing_aa.ht",
+        _read_if_exists=not overwrite_temp,
+        overwrite=overwrite_temp,
+    )
+    miss_start_stop_transcripts = miss_start_stop_ht.aggregate(
+        hl.agg.collect_as_set(miss_start_stop_ht.transcript)
+    )
+    filt_context_ht = get_aa_from_context(
+        overwrite_temp=False, keep_transcripts=miss_start_stop_transcripts
+    )
+    filt_context_ht = filt_context_ht.filter(
+        hl.literal(miss_start_stop_transcripts).contains(filt_context_ht.transcript)
+    )
+    filt_context_ht = filt_context_ht.group_by("transcript").aggregate(
+        max_aa_num=hl.agg.max(filt_context_ht.protein_start)
+    )
+    filt_context_ht = filt_context_ht.checkpoint(
+        f"{TEMP_PATH_WITH_FAST_DEL}/rmc/transcript_start_stop_missing_max_aa_num.ht",
+        _read_if_exists=not overwrite_temp,
+        overwrite=overwrite_temp,
+    )
+    miss_start_stop_ht = miss_start_stop_ht.annotate(
+        **filt_context_ht[miss_start_stop_ht.transcript]
+    )
+    miss_start_stop_ht = miss_start_stop_ht.annotate(
+        start_aa=hl.if_else(
+            hl.is_missing(miss_start_stop_ht.start_aa),
+            hl.if_else(
+                # Set all missing + strand transcript starts to Met1
+                miss_start_stop_ht.strand == "+",
+                "Met1",
+                hl.format("Ter%s", miss_start_stop_ht.max_aa_num),
+            ),
+            miss_start_stop_ht.start_aa,
+        ),
+        stop_aa=hl.if_else(
+            hl.is_missing(miss_start_stop_ht.stop_aa),
+            hl.if_else(
+                # Set all missing - strand transcript stops to Met1
+                miss_start_stop_ht.strand == "-",
+                "Met1",
+                hl.format("Ter%s", miss_start_stop_ht.max_aa_num),
+            ),
+            miss_start_stop_ht.start_aa,
+        ),
+    )
+    return miss_start_stop_ht.select("start_aa", "stop_aa")
+
+
+def fix_region_start_stops(
+    ht: hl.Table, context_ht: hl.Table, overwrite_temp: bool
+) -> hl.Table:
+    """
+    Fix missing start or stop amino acid information for non-transcript starts and stops.
+
+    :param ht: Input HT. Should be RMC results HT annotated with amino acid information.
+    :param overwrite_temp: Whether to overwrite temporary data.
+        If False, will read existing temp data rather than overwriting.
+        If True, will overwrite temp data.
+    :return: HT containing start and stop amino acids only for RMC regions that were
+        previously missing amino acid information.
+    """
+    missing_ht = ht.filter(hl.is_missing(ht.start_aa) | hl.is_missing(ht.stop_aa))
+    missing_transcripts = missing_ht.aggregate(
+        hl.agg.collect_as_set(missing_ht.transcript)
+    )
+    filt_context_ht = context_ht.filter(
```

Comment:
This table isn't used

Reply:
wait, it is, right? at the end

Reply:
Oh you're right, it was marked as unused in my vscode and I must have not scrolled down far enough to check the whole function. Thanks for correcting me!